### PR TITLE
add dockerfile

### PR DESCRIPTION
**Pull request by siddontang**
### What problem does this PR solve? 

Support Docker

### What is changed and how it works?

Add dockerfile to build docker image

```
docker build -t pingcap/dm .
```

Tests 

 - Manual test (add detailed scripts or steps below)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -11,7 +11,6 @@ GOTEST   := CGO_ENABLED=1 $(GO) test
 PACKAGES  := $$(go list ./... | grep -vE 'tests|cmd|vendor|pbmock')
 FILES    := $$(find . -name "*.go" | grep -vE "vendor")
 TOPDIRS  := $$(ls -d */ | grep -vE "vendor")
-SHELL    := /usr/bin/env bash
```
Comment (siddontang) [inline Makefile]: Alpine doesn't have bash by default, and we don't use this too, so here just remove.

**Comment by siddontang** [comment]
may provide an example about how to use all in Docker compose later.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+COPY go.mod .
+COPY go.sum .
+
+RUN GO111MODULE=on go mod download
+
+COPY . .
+
+RUN make dm-worker dm-master dm-tracer dmctl
+
+FROM alpine:3.5
```
Comment (csuzhangxc) [inline Dockerfile]: How about using `alpine:3.10` (resued from  `golang:1.13-alpine`, ref  https://github.com/docker-library/golang/blob/f30f0428221b94c7dcb414ebdcea83106df20897/1.13/alpine3.10/Dockerfile)

Comment (siddontang) [reply Dockerfile]: @iamxy 

which Apline version do you use now?

Comment (iamxy) [reply Dockerfile]: I prefer `alpine:3.10`.

Comment (tennix) [reply Dockerfile]: The current base image used by our official images is pingcap/alpine-glibc. It's still 3.8 not 3.10.
I think we need to update the base image to 3.10. /cc @qrr1995

Comment (siddontang) [reply Dockerfile]: so let's use 3.10 here.

Comment (qrr1995) [reply Dockerfile]: The `pingcap/alpine-glibc:latest` image  has been updated to alpine-3.10 now, just check it   with 
```
$ docker pull pingcap/alpine-glibc
$ docker run pingcap/alpine-glibc cat /etc/alpine-release
```

**Comment by GregoryIan** [comment]
please fix CI @siddontang

**Pull request merged**